main.go

```diff
@@ -177,23 +184,17 @@ func mapCRDAvailability(crdNames ...string) map[string]bool {
 
 	CRDExist := map[string]bool{}
 
-	for _, crdName := range crdNames {
-		crd := apiextensionsv1.CustomResourceDefinition{}
-		err = k8sClient.Get(context.Background(), types.NamespacedName{Name: crdName}, &crd)
+	crds := apiextensionsv1.CustomResourceDefinitionList{}
+	err = k8sClient.List(context.Background(), &crds)
 
-		if err != nil && !errors.IsNotFound(err) {
-			setupLog.Error(err, fmt.Sprintf("Error Getting CRD for %s", crdName))
-			os.Exit(1)
-		}
+	if err != nil && !errors.IsNotFound(err) {
+		setupLog.Error(err, fmt.Sprintf("Error Getting CRDs installed on the cluster"))
+		os.Exit(1)
+	}
 
-		CRDExist[crdName] = crd.UID != ""
+	for _, crd := range crds.Items {
+		CRDExist[crd.Name] = true
```

Comment:
This way of looping over a list will create a copy of each CRD struct on that list as it is passed by value and not by reference. Please use the following instead
```suggestion
	for i := range crds.Items {
		CRDExist[crds.Items[i].Name] = true
```